kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaAuthnHandler.java

```diff
@@ -742,7 +747,7 @@ else if (offsetFetchRequestData.topics() != null) {
                         .build(apiVersion);
                 break;
             case UPDATE_METADATA:
-                req = new UpdateFeaturesRequest((UpdateFeaturesRequestData) reqBody, apiVersion);
+                req = new UpdateFeaturesRequest((UpdateFeaturesRequestData) reqBody, apiVersion); // TODO: check why
```

Comment:
If you mean, "why it's not using the builder", I'm not sure there is a reason. Happy to switch to using the builder if it doesn't fail the tests.

Reply:
When I first read it, I didn't understand why the `UPDATE_METADATA` returns an `UpdateFeaturesRequest` rather than an `UpdateMetadataRequest` ? Is it voluntary ?